File: plugins/togetherai-skills/skills/together-chat-completions/scripts/debug_headers.py
#!/usr/bin/env python3
"""
Together AI Chat Completions — Debug Headers and Raw Responses (v2 SDK)

Inspect parsed chat output together with raw response headers for latency, routing,
and rate-limit debugging.

Usage:
    python debug_headers.py

Requires:
    uv pip install "together>=2.0.0"
    export TOGETHER_API_KEY=your_key
"""

from together import Together

client = Together()


def main() -> None:
    """Print a parsed chat response together with selected response headers."""
    response = client.chat.completions.with_raw_response.create(
        model="openai/gpt-oss-20b",
        messages=[{"role": "user", "content": "Say hello in one sentence."}],
        extra_headers={"x-together-debug": "1"},
    )

    parsed = response.parse()
    print("=== Parsed Response ===")
    print(parsed.choices[0].message.content)
    print()

    print("=== Selected Headers ===")
    interesting_headers = [
        "x-request-id",
        "x-together-traceid",
        "x-cluster",
        "x-engine-pod",
        "x-api-received",
        "x-api-call-start",
        "x-api-call-end",
        "x-inference-version",
        "x-ratelimit-limit",
        "x-ratelimit-remaining",
        "x-ratelimit-reset",
        "x-tokenlimit-limit",
        "x-tokenlimit-remaining",
    ]
    header_map = dict(response.headers)
    for key in interesting_headers:
        if key in header_map:
            print(f"{key}: {header_map[key]}")


if __name__ == "__main__":
    main()

File: plugins/togetherai-skills/skills/together-chat-completions/scripts/debug_headers.ts
#!/usr/bin/env -S npx tsx
/**
 * Together AI Chat Completions — Debug Headers and Raw Responses
 *
 * Inspect parsed chat output together with raw response headers for latency,
 * routing, and rate-limit debugging.
 *
 * Usage:
 *   npx tsx debug_headers.ts
 *
 * Requires:
 *   npm install together-ai
 *   export TOGETHER_API_KEY=your_key
 */

import Together from "together-ai";

const client = new Together({
  apiKey: process.env.TOGETHER_API_KEY,
});

async function main(): Promise<void> {
  const response = await client.chat.completions.create(
    {
      model: "openai/gpt-oss-20b",
      messages: [{ role: "user", content: "Say hello in one sentence." }],
    },
    { headers: { "x-together-debug": "1" } }
  ).asResponse();

  const parsed = await response.json();
  console.log("=== Parsed Response ===");
  console.log(parsed.choices[0].message.content);
  console.log();

  console.log("=== Selected Headers ===");
  const interestingHeaders = [
    "x-request-id",
    "x-together-traceid",
    "x-cluster",
    "x-engine-pod",
    "x-api-received",
    "x-api-call-start",
    "x-api-call-end",
    "x-inference-version",
    "x-ratelimit-limit",
    "x-ratelimit-remaining",
    "x-ratelimit-reset",
    "x-tokenlimit-limit",
    "x-tokenlimit-remaining",
  ];

  for (const key of interestingHeaders) {
    const value = response.headers.get(key);
    if (value) {
      console.log(`${key}: ${value}`);
    }
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
